### ISSUE TITLE
Annotating Custom Data

### ISSUE THREAD

**Issue by MCemGunes** [daerduoCarey/partnet_anno_system#23]
Hello, I’m Cem Gunes

I have questions about labeling my custom models. I have read the paper, checked other closed issues; however, I could not apply a clear pipeline to prepare my models as parts of an assembly. Since I am out of the computer science community, I might be skipping some basic steps. Thanks for your understanding. 

Firstly, I want to clarify what I aim to do. As an architect and computational designer, I want to prepare point cloud or mesh models of buildings as parts of a whole and train some deep learning models to see part segmentation performance to evaluate part-whole relations. So I created my custom mesh models using Rhinoceros 3D software and exported the 3d mesh models (.obj) with 4 labels. Likely, I will experiment with hierarchical tree-hier structures to identify sub-elements in further steps. To illustrate, I can segment the wall then identify the bricks. Nevertheless, I would be glad to create a dataset using the labels in the same hierarchy levels.   

After some research, this annotator tool seems promising, but I could not manage to import my models and annotate them. Also, I have some problems using the sample models.

The main problem here is I could not even visualize the sample data properly except for one of them. 

I see only one ID of the chair models. When I try to annotate it, I can not pick any specific point, so I can not cut the model as I wanted. 

![Screenshot from 2022-01-06 19-15-33](https://user-images.githubusercontent.com/25433512/148449417-8a00fea9-afff-4272-84d8-bf5ad6130f48.png)
![Screenshot from 2022-01-06 19-16-06](https://user-images.githubusercontent.com/25433512/148449640-69b5a75b-e2b4-40b4-befa-25f80408b4d9.png)

![Screenshot from 2022-01-06 19-16-40](https://user-images.githubusercontent.com/25433512/148449687-208e17f4-a046-4a76-9cb5-682d51823b0d.png)
![Screenshot from 2022-01-06 19-16-57](https://user-images.githubusercontent.com/25433512/148449690-78fb77f1-b55f-4b91-b747-2871ca66ecde.png)

When I push ‘s’ it selects the whole part, I can not use ‘c' cut command. Moreover, requesting remeshing 'm' returns an error.
You can see the backend and frontend errors when I try to re-mesh the model.

![Screenshot from 2022-01-06 19-10-07](https://user-images.githubusercontent.com/25433512/148449510-714a2ac3-992c-4a03-9f19-de39d7289076.png)
![Screenshot from 2022-01-06 19-10-42](https://user-images.githubusercontent.com/25433512/148449544-15aad959-f509-42c9-8a42-074f5d4aaa60.png)
![Screenshot from 2022-01-06 19-14-51](https://user-images.githubusercontent.com/25433512/148449554-9bfb9002-7733-403e-bb14-0177e3815b77.png)
![Screenshot from 2022-01-06 19-19-05a](https://user-images.githubusercontent.com/25433512/148451761-98af7eca-bb2d-4bfa-8c2c-7924f436fae2.png)
![Screenshot from 2022-01-06 19-18-53a](https://user-images.githubusercontent.com/25433512/148451934-16e8b83b-6e0f-420e-8e5c-e8a9742e1e3e.png)

So I can not figure out why I can not see all the sample models you provided in the ShapeNet format and why I can not annotate them properly. 

Finally, if it is genuinely possible for other researchers to annotate their custom models and use the format you have created, please explain it to me step by step from scratch. It seems like we can not transform our data to the ShapeNet format. So it seems like there is a strict limitation. Please inform me if I’m wrong. If somebody has already tried and succeeded, I am totally open to any guidance. 

Here is my system 
I use Ubuntu 20.04.3 LTS focal. (amd64)
I installed MySQL 5.7.36, as mentioned in the instructions.
I installed Manifold and defined it as Globally executable in the `model-fixer` name.
I installed npm ( 8.3.0) and node (17.3.0)
Copied geometry_utils.py to “/partnet_anno_system/storage/data”
Installed, mysql.connector, progressbar

server/config/server.js
```
// SET UP HERE!
server.HOST = 'http://localhost';
server.PORT = 3000;
server.DB_HOST = "localhost";
server.DB_USER = "mucemgunes";
server.DB_PASSWORD = "****";
server.DB_NAME = "partnet_anno_system";
server.CODE_DIR = "/home/ubuntuoslo/git_repo/partnet_anno___/partnet_ubuntu/partnet_anno_system"
```

client/config/backend.js
```
// Host and Port Information (SET UP HERE!)
be_config.remoteHost = 'http://localhost';
be_config.remotePort = '3000';

```

I can provide any additional information if required. 

Thanks for your immense documentation and publishing this work,

Greetings

**Comment by Mdemerdash** [comment]
@MCemGunes Hi, were you able to figure out this pipeline?

**Comment by maxerbubba** [comment]
> @maxerbubba Hi, were you able to figure out this pipeline?

I don’t know anything I don’t work on this project

**Comment by Mdemerdash** [comment]
> > @maxerbubba Hi, were you able to figure out this pipeline?
> 
> I don’t know anything I don’t work on this project

Sorry for the inconvenience, i mentioned you by mistake

**Comment by MCemGunes** [comment]
@Mdemerdash  Hi! Sorry for the late response, unfortunately, I gave up after several tries.

**Comment by Mdemerdash** [comment]
Hi @MCemGunes thank you for your reply. were you able to find similar alternatives? I'm about to give up too :D

**Comment by MCemGunes** [comment]
@Mdemerdash No, not really, I changed my method to classical semantic segmentation without part hierarchies. :)

**Comment by Mdemerdash** [comment]
@MCemGunes that's sad..I'm actually making some progress..now able to cur the parts in the anno system. I also created a new category where i actually used some files of the chair category that i can produce on my own and I'm able to annotate my "own" chair category. If you still have some slight interest feel free to drop me an email. mdemerdash89@hotmail.com

**Comment by MCemGunes** [comment]
Thank you for being so friendly and helpful. Nowadays I'm busy with other tasks, but when I'm free I am definitely going to check it again and let you know if I got stuck in one of the steps.
